### DISCUSSION TITLE
GitHub Issues Caching & User Experience Feedback

### DISCUSSION THREAD

**Discussion by 0x4007** [ubiquity/work.ubq.fi#11] [Show and tell]
Hey there! I manually created a relatively complex caching system that takes into account the [devpool-directory's "mirror"](https://github.com/ubiquity/devpool-directory/issues) of issues. I would like your feedback around the reliability of the caching upon normal use (it seems to not do well when I abuse it for testing due to rate limits regarding the GitHub API and having to log into GitHub.)

Beyond that, I'm interested to receive general feedback on the user experience (i.e. usability) of the interface. I'm trying to make a great first impression here for new would-be contributors. We are also funneling all of the candidates we are in talks with for the new office to the DevPool to serve as a "paid technical interview" so these impressions will be tangible as we meet in person with these people. 

The latest deployment as of this moment is https://e39078d5.devpool-directory-ui.pages.dev/ cf3dae4b8da99853c523d5bcdb5ee56a5ee9a58b

The app can normally be accessed at https://work.ubq.fi/

Thanks!

**Comment by 0x4007** [comment]
P.S. there is also secret keyboard navigation once you preview some issues (easier to get the full experience when logged in, as rate limits are increased above 60 requests.) 

I tried to make this as usable as possible for when we have tons of issues to sort though. 

I know its not perfect but I hope that the minor hiccups are not deal breakers for fresh eyes. Anyways, that is the purpose of this "show and tell" is to get fresh eyes on it for feedback.

**Comment by rndquu** [comment]
Works fine, I do miss the labels like "solidity", "devops", etc...but I guess we need to set them first in the [devpool issues](https://github.com/ubiquity/devpool-directory/issues) on github

**Reply by 0x4007** [reply]
If you check on desktop, the rows should make a special accommodation for a single extra label to accommodate this exact scenario. Unfortunately the strategy is a bit random, it just takes the first label it can find that is not one of the special assistive pricing labels. 


I'll attach a screenshot later when I get to my computer.

**Reply by 0x4007** [reply]
<img width="647" alt="Screenshot 2024-01-31 at 17 40 41" src="https://github.com/ubiquity/work.ubq.fi/assets/4975670/7f6a4504-5b52-43c0-9b2d-398689e1ab97">

**Comment by gitcoindev** [comment]
I add my thumbs up. Tested on Linux / Brave browser and on mobile Android / Chrome. UI looks clean, filtering and sorting works fine. Markdown rendering looks subjectively nice to my eyes and encourages me to execute the `/start` command. I tested all hyperlinks on a few tasks, no issues observed.

**Reply by 0x4007** [reply]
> and encourages me to execute the /start command.

I don't understand what you mean by this.

**Reply by molecula451** [reply]
haha, more like the UI it's so nice that it delights em to /start a task!

**Reply by gitcoindev** [reply]
@pavlovcik yes @molecula451 was right -)

**Comment by molecula451** [comment]
very intuitive and resonates with the Ubiquity ideal, not sure if we planning to add the Dark Mode it can be easily be re-implemented from the Dollar UI, to have both feelings, but this is just cosmetic, the important here is the UI fetchs de available tasks in a very sophisticated way

**Reply by 0x4007** [reply]
I tried to be deliberate about making the UI inputs streamlined to reduce clutter. It reads from your OS "prefers light mode" or dark mode settings. 

![image](https://github.com/ubiquity/work.ubq.fi/assets/4975670/f64bf568-687c-406e-867c-53ea64578b15)

![image](https://github.com/ubiquity/work.ubq.fi/assets/4975670/f5241ea4-d2f8-435e-807d-7bcfbe738269)

I will take this into consideration for adding another UI element for manually toggling this.
